### PR TITLE
feat(SSG): RHICOMPL-1736 record rule order for each imported profile

### PR DESCRIPTION
**Pull request by skateman**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/services/concerns/xccdf/profile_rules.rb [hunk 1]
```diff
@@ -7,26 +7,29 @@ module ProfileRules
 
     included do
       def save_profile_rules
-        ::ProfileRule.transaction do
```
Comment (skateman) [inline app/services/concerns/xccdf/profile_rules.rb:10]: I feel like this is no longer necessary after https://github.com/RedHatInsights/compliance-backend/pull/1150 cc @vkrizan

File: app/services/concerns/xccdf/rules.rb [hunk 2]
```diff
@@ -7,25 +7,18 @@ module Rules
...
+        # This is needed for future computations with rule identifiers
+        @new_rules = @rules.select(&:new_record?)
 
-      def old_rules
-        split_rules.last
+        ::Rule.import!(@rules,
+                       on_duplicate_key_update: {
+                         conflict_target: %i[ref_id benchmark_id],
+                         columns: %i[precedence]
+                       }, ignore: true)
```
Comment (vkrizan) [inline app/services/concerns/xccdf/rules.rb:21]: `ignore: true` should be dropped, preceded by `on_duplicate_key_update`.

File: app/services/concerns/xccdf/rules.rb [hunk 3]
```diff
@@ -7,25 +7,18 @@ module Rules
...
-
-      def split_rules
-        @split_rules ||= @rules.partition(&:new_record?)
-      end
-
-      def new_rules
-        @new_rules ||= split_rules.first
-      end
+        # This is needed for future computations with rule identifiers
+        @new_rules = @rules.select(&:new_record?)
```
Comment (vkrizan) [inline app/services/concerns/xccdf/rules.rb:15]: Hopefully this would not forget to recompute precedence when just new rules would be used.

**Review by marleystipich2 — APPROVED** [review]
Looks good! Thanks, @skateman!

**Pull request merged**